lib/tasks/keel.rake

```diff
@@ -1,246 +1,246 @@
-		task :setup => [:check_gcloud_executables, 
-										:check_configuration,
-										:update_tools,
-										:gcloud_auth,
-										:set_gcloud_properties,
-										:install_k8s,
-										:authenticate_k8s] 
-
-		task :check_gcloud_executables do
-			config    = Keel::GCloud::Config.new
-
-			if config.executable_missing?
-				message = 'Install Google Cloud command line tools'
-				message += "\n"
-				message += 'See: https://cloud.google.com/sdk/'
-
-				abort message.red
-			end
-		end
-
-		task :check_configuration do
-			config    = Keel::GCloud::Config.new
-
-			if config.system_configured?
-				abort 'App appears to already be configured on your system.'.green
-			end
-		end
-
-		task :install_k8s do
-			prompter  = Keel::GCloud::Prompter.new
-			prompter.print 'Install Kubernetes', :info
-			Keel::GCloud::Component.install_k8s
-		end
-
-		task :authenticate_k8s do
-			prompter  = Keel::GCloud::Prompter.new
-			prompter.print 'Pulling Kubernetes auth configuration', :info
-			auth = Keel::GCloud::Auth.new config: config
-			auth.authenticate_k8s
-		end
-
-		task :gcloud_auth do
-			prompter  = Keel::GCloud::Prompter.new
-			prompter.print 'Authenticating with Google Cloud', :info
-			Keel::GCloud::Auth.authenticate
-		end
-
-		task :update_tools do
-			prompter  = Keel::GCloud::Prompter.new
-			prompter.print 'Updating tools', :info
-			Keel::GCloud::Component.update
-		end
-
-		task :set_gcloud_properties do 
-			config    = Keel::GCloud::Config.new
-			prompter  = Keel::GCloud::Prompter.new
-			prompter.print 'Setting gcloud properties', :info
-			config.set_properties
-		end
-
-		desc 'Pulls logs for a given environment'
-		task :logs, [:environment] do |_, args|
-			prompter  = Keel::GCloud::Prompter.new
-			config    = Keel::GCloud::Config.new
-			app       = config.app_name
-
-			# Fetch namespaces from k8s
-			namespaces = Keel::GCloud::Kubernetes::Namespace.fetch_all
-			unless namespaces
-				message = 'Unable to connect to Kubernetes, please try again later...'
-				prompter.print message, :error
-				abort
-			end
-
-			# Prompt the user for the env and to log and whether to tail the logs
-			deploy_env    = prompter.prompt_for_namespace namespaces, args[:environment]
-			tail          = prompter.prompt_for_tailing_logs
-
-			prompter.print "Getting pod information for #{deploy_env}", :info
-			pods = Keel::GCloud::Kubernetes::Pod.fetch_all deploy_env, app
-
-			unless pods.length > 0
-				message = "Could not find pods in the \"#{deploy_env}\" environment."
-				prompter.print message, :error
-				abort
-			end
-
-			# It seems that the first pod is the one we really want to look at for logs
-			pod = pods.first
-			pod.logs tail
-		end
+namespace :keel do
+  def pick_sha(sha)
+    unless @sha
+      prompter  = Keel::GCloud::Prompter.new
+      @sha = prompter.prompt_for_sha sha
+    end
+    @sha
+  end
+
+  def pick_namespace(namespace)
+    unless @namespace
```

Comment:
same question about `@namespace`, doesn't seem to do anything to check for an existing instance variable.